internal/controllers/core/tiltfile/api.go

```diff
@@ -348,6 +392,15 @@ func updateNewObjects(ctx context.Context, client ctrlclient.Client, newObjects,
 			labelsChanged := !apicmp.DeepEqual(old.GetLabels(), obj.GetLabels())
 			if specChanged || labelsChanged {
 				obj.SetResourceVersion(old.GetResourceVersion())
+				if cm, ok := obj.(*v1alpha1.ConfigMap); ok {
+					// Tiltfiles can create ConfigMaps with default values, but
+					// they shouldn't blow away values that were modified elsewhere.
+					// At some point, we might have a use case that contradicts this and
+					// need to make this more complicated.
```

Comment:
Ah, nice - I feel like this will be important for things like trigger mode as well (since currently after toggling via UI it'll get reset next Tiltfile execution). Perhaps `Patch()` / SMD would be the more general solution here in the future